airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py

```diff
@@ -32,11 +33,17 @@
 from airflow.models import BaseOperator
 from airflow.providers.amazon.aws.hooks.dynamodb import DynamoDBHook
 from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.utils.types import NOTSET, ArgNotSet
 
 if TYPE_CHECKING:
     from airflow.utils.context import Context
 
 
+_DEPRECATION_MSG = (
+    "The aws_conn_id parameter has been deprecated. You should pass instead the source_aws_conn_id parameter."
```

Comment:
```suggestion
    "The aws_conn_id parameter has been deprecated. Use the source_aws_conn_id parameter instead."
```

I'm still not convinced that dropping aws_conn_id is the right answer here and would prefer to see it left in, but if consensus is to drop it then I guess source_ is a suitable name.

Reply:
I think as we are not dropping it and the new code matches the behaviour of the old one if you only use `aws_conn_id`, just adding a deprecation warning, we are not twisting anyones hands too much. 
On the other side, keeping all 3 connection parameters while only using maximum 2 out of them is not very clean, IMO.